scalanet/src/io/iohk/scalanet/peergroup/kademlia/KNetwork.scala

```diff
@@ -0,0 +1,52 @@
+package io.iohk.scalanet.peergroup.kademlia
+
+import io.iohk.scalanet.peergroup.kademlia.KMessage.KRequest
+import io.iohk.scalanet.peergroup.kademlia.KMessage.KRequest.FindNodes
+import io.iohk.scalanet.peergroup.kademlia.KMessage.KResponse.Nodes
+import io.iohk.scalanet.peergroup.{Channel, PeerGroup}
+import monix.execution.Scheduler
+import monix.reactive.Observable
+import org.slf4j.LoggerFactory
+
+import scala.concurrent.Future
+import scala.util.Failure
+
+trait KNetwork[A] {
+  def server: Observable[(Channel[A, KMessage[A]], KRequest[A])]
```

Comment:
is it correct to have the same type parameter for both the `Channel` address and the `KMessage`?
I would think that the Channel address type could be, for example, a `BitVector` representing a node ID.

Reply:
Pls see the latest version which simplifies this. When I came to write the test app, some of the abstraction was difficult to understand and served little point.